### ISSUE TITLE
Marlin 2.0.7.2

### ISSUE THREAD

**Issue by anteyru** [makerbase-mks/MKS-TOOL#9]
The current version of Marlene is already out of date. Planning to update Marlin to version 2.0.7.2?
Also, there is no UART support for Robin Nano 2.

**Comment by mks-viva** [comment]
Hi! If you download offical marlin 2.0.7.2 and MKS Robin Nano V2 uart mode iwork is ok?

**Comment by anteyru** [comment]
Yes, everything works fine. Therefore, I am surprised that you have a very old version.

**Comment by mks-viva** [comment]
MKS TOOL may not be updated yet, it is recommended that you directly use the official firmware.

**Comment by anteyru** [comment]
Marlin 2.0.7.2
![20201029_110844](https://user-images.githubusercontent.com/8513289/99954694-399e0880-2d94-11eb-93ef-e5b60cf381aa.jpg)

**Comment by mks-viva** [comment]
Wa, look it is very ok!

**Comment by anteyru** [comment]
Firmware from MKSTOOL https://www.baizhongyun.cn/home/mkstoolview
![20201122_193557](https://user-images.githubusercontent.com/8513289/99954927-8eda1a00-2d94-11eb-8a5c-a80f1c5e71cc.jpg)
![20201122_193607](https://user-images.githubusercontent.com/8513289/99955152-e4aec200-2d94-11eb-8a86-86976bf2af38.jpg)

**Comment by mks-viva** [comment]
This looks a bit bad.

**Comment by anteyru** [comment]
There is no uart in the official firmware, the motors turn in the other direction.

**Comment by mks-viva** [comment]
Uart settings is easy.
1. In Configuration.h file set driver stype as: TMC2209
(If use TMC2208 or TMC2225, you can set it directly)

**Comment by anteyru** [comment]
I'm talking about the standard firmware.
makerbase-mks/MKS-Robin-Nano-V2.X/Firmware/Robin-nano-V2.0-Marlin-bugfix2.0.x-MKS-2.1.1.rar

**Comment by mks-viva** [comment]
Can you provide a vodies for me when you use makerbase-mks/MKS-Robin-Nano-V2.X/Firmware/Robin-nano-V2.0-Marlin-bugfix2.0.x-MKS-2.1.1.rar firmware？

**Comment by anteyru** [comment]
> Can you provide a vodies for me when you use makerbase-mks/MKS-Robin-Nano-V2.X/Firmware/Robin-nano-V2.0-Marlin-bugfix2.0.x-MKS-2.1.1.rar firmware？
Home and zero to the left near corner.
Clicking on "+" moves to 0.
https://www.youtube.com/watch?v=TbBSxuNS5eA

**Comment by anteyru** [comment]
[Robin Nano v2 motor direct](https://www.youtube.com/watch?v=TbBSxuNS5eA)

**Comment by mks-viva** [comment]
> > Can you provide a vodies for me when you use makerbase-mks/MKS-Robin-Nano-V2.X/Firmware/Robin-nano-V2.0-Marlin-bugfix2.0.x-MKS-2.1.1.rar firmware？
> > Home and zero to the left near corner.
> > Clicking on "+" moves to 0.
> > There is no way to change direction from the menu.
> > If you collect through the site, the firmware is configured. But the pictures are not loaded.
> > [MKS-Robin-Nano-V2 motor direct](https://www.youtube.com/watch?v=TbBSxuNS5eA)

You need to modify the direction parameter in Configuration.h file and recompile and reupdate it

**Comment by anteyru** [comment]
> > > Can you provide a vodies for me when you use makerbase-mks/MKS-Robin-Nano-V2.X/Firmware/Robin-nano-V2.0-Marlin-bugfix2.0.x-MKS-2.1.1.rar firmware？
> > > Home and zero to the left near corner.
> > > Clicking on "+" moves to 0.
> > > There is no way to change direction from the menu.
> > > If you collect through the site, the firmware is configured. But the pictures are not loaded.
> > > [MKS-Robin-Nano-V2 motor direct](https://www.youtube.com/watch?v=TbBSxuNS5eA)
> 
> You need to modify the direction parameter in Configuration.h file and recompile and reupdate it

Ok, but then how do I add the GUI from ronin nano v2 to my marlin?

**Comment by mks-viva** [comment]
Offical Marlin has add MKS UI for MKS Robin Nano V2.

**Comment by anteyru** [comment]
> Offical Marlin has add MKS UI for MKS Robin Nano V2.

> // TFT LVGL UI
//
// Using default MKS icons and fonts from: https://git.io/JJvzK
// Just copy the 'assets' folder from the build directory to the
// root of your SD card, together with the compiled firmware.
//
//#define TFT_LVGL_UI_FSMC  // Robin nano v1.2 uses FSMC
#define TFT_LVGL_UI_SPI   // Robin nano v2.0 uses SPI

**Comment by anteyru** [comment]
Is it about this?

**Comment by mks-viva** [comment]
The latest version Marlin-bugfix-2.0.x . If you want use MKS UI, you need set it:
Platformio.ini file: 
```
    Set default_envs = mks_robin_nano35
```
Configuration.h file: 
```
    #define SERIAL_PORT 3
    #define MOTHERBOARD BOARD_MKS_ROBIN_NANO_V2
    #define MKS_TS35_V2_0
    #define TFT_LVGL_UI
    #define TOUCH_SCREEN
    #define SDSUPPORT
```

**Comment by tizi24** [comment]
is it compatible with MKS TS24 V2.0?
